### PR TITLE
Allow plugins to suppress entity outputs (fix 5515).

### PR DESCRIPTION
**Pull request by klausenbusk**
https://bugs.alliedmods.net/show_bug.cgi?id=5515
It seems like this bug/patch has been forgotten, so I opened a pull request.
This is just Peace-Maker patch, where I updated the changes in the include file to the new syntax.

Kristian

### CODE REVIEW

**Comment by klausenbusk** [comment]
Hmm. I did something wrong in the .inc file.
```
../../plugins/include/sdktools_entoutput.inc(52) : error 120: methodmap and class signatures must use new-style type declarations
```

File: plugins/include/sdktools_entoutput.inc [hunk 1]
```diff
@@ -42,9 +42,15 @@
...
- * @noreturn
+ * @return					Anything other than Plugin_Continue will supress this event,
+ *								returning Plugin_Continue will allow it to propagate the results
+ *								of this output to any entity inputs.
  */
-typedef EntityOutput = function void (const char[] output, int caller, int activator, float delay);
+typeset EntityOutput
+{
+	function void (const char[] output, int caller, int activator, float delay);
+	function Action (const String:output[], int caller, int activator, float delay);
```
Comment (psychonic) [inline plugins/include/sdktools_entoutput.inc]: This is the line causing your error. (String)

Comment (klausenbusk) [inline plugins/include/sdktools_entoutput.inc]: Fixed (I hope).

**Comment by klausenbusk** [comment]
What do you think? Do the code need any change?

**Comment by psychonic** [comment]
Please don't bump Pull Requests. It's only been a few hours.

**Comment by psychonic** [comment]
Thanks for helping push this forward.

**Comment by klausenbusk** [comment]
Can I open a pr for 1.7?

**Comment by psychonic** [comment]
PR's aren't necessary for cherry-picking changes to another branch unless large further changes are required to get it to merge due to structural changes between branches. I don't think that's the case here - we can do a normal cherry-pick when ready.

However, since it *just* landed, and since 1.8 is near stable anyway, I'm not sure that we'll opt to do so.

**Comment by klausenbusk** [comment]
I see. I just wait for 1.8 then. I need it for a not mandatory plugin feature, I can code it some other way, but this is most correct way.

**Pull request merged**